client/src/components/Map.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { Component, useRef, useState, useEffect } from 'react';
+import * as d3 from 'd3';
+import ZoomableSVG from './ZoomableSVG';
+
+type Prop = {
+    layoutImage: string
+}
+
+export default class Map extends Component<Prop>{
+
+    mapRef!: SVGSVGElement
+
+    private drawMap(){
+
+        const svgElement = d3.select(this.mapRef)
+        .append("image")
+        .attr("xlink:href", `/submaps/${this.props.layoutImage}.svg`)
```

Comment:
This line adds a number of assumptions:
- the image is located in submaps folder
- its name ends with .svg

When passing an image path to the component via props the caller of the component should define the whole path. I see you adding these assumption in other places, too. Like in DirectionsCard.

This line should be as follows:
`.attr("xlink:href", this.props.layoutImage)`